lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -185,6 +186,8 @@ ExportContext ExportContext::forDeclSignature(Decl *D) {
   bool spi = Ctx.LangOpts.LibraryLevel == LibraryLevel::SPI;
   bool implicit = false;
   bool deprecated = false;
+  // no-async is only computed locally
+  bool noAsync = D->getAttrs().getNoAsync(Ctx) == nullptr;
```

Comment:
double-check this. The naming looks like it might be flipped?

Reply:
The result of this and `ExportContext::NoAsync` doesn't appear to be used anywhere.
The attribute is explicitly checked in `diagnoseDeclUnavailableFromAsync` in `TypeCheckAvailability.cpp`.
This can probably get removed.